cocos/core/platform/event-manager/input-manager.ts

```diff
@@ -173,7 +173,7 @@ class InputManager {
         }
         if (handleTouches.length > 0) {
             // this._glView!._convertTouchesWithScale(handleTouches);
-            const touchEvent = new EventTouch(handleTouches, false, EventTouch.MOVED, this._getUsefulTouches());
+            const touchEvent = new EventTouch(handleTouches, false, EventTouch.MOVED, macro.ENABLE_MULTI_TOUCH ? this._getUsefulTouches() : [handleTouches[0]]);
```

Comment:
Obviously, when you made four times the same logic and they are all using the same function, you should move that logic into the function.

Why not just return correct touch array in `_getUsefulTouches`, by the way, it also avoids another array creation

Reply:
I don’t think it’s appropriate to add some extraneous parameters to the function. The modification avoids the creation of another array.

Reply:
There is no extra parameter needed, just use `macro.ENABLE_MULTI_TOUCH`, it's a global setting, so I think it's appropriate to make the touch collection process obey the behavior